entrypoint.sh

```diff
@@ -19,46 +31,45 @@ process_input(){
                 scan_file "$path"
             fi
         done
-        if [ -z "$execution_mode" ]; then exit $status_code; fi
+        [ -z "$execution_mode" ] && exit $status_code
     else
         scan_all "$my_dir"
-        if [ -z "$execution_mode" ]; then exit $status_code; fi
+        [ -z "$execution_mode" ] && exit $status_code
     fi
 }
 
 scan_file(){
     local file_path=$1
     local file=$(basename -- "$file_path")
-    local extension="${file##*.}"
-
-    if [ "$extension" == "sh" ]; then
+    local first_line=$(head -n 1 "$file_path")
+    if [[ -x "$file_path" ]] && [[ "$first_line" == "#!"* ]]; then
```

Comment:
`-x "$file_path"`

Why are we checking if a file is executable? Would the `#!` check be enough

Context: I had a few shell scripts that weren't executable, so I was just wondering. Also, I made a new file `temp` without an extension, made it executable and did not add the shebang. Shellcheck reported an error. So I think the shebang should be enough